Update Lucene version to 9.11

benwtrent opened the pull request.

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-1768b2061ba

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-e51ca16de96

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-81a7c405875

jpountz committed:
Fix compilation.

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-caf168e6e49

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-caf168e6e49

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-0ad98cc4a9e

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

ChrisHegarty committed:
Add 9_11 IndexVersion

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-0ab29c695c0

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-6c8206686de

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-6c8206686de

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-e5058e73baf

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-cbea492bf0a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-5683676765c

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-8a555ebdc3c

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-1883c10717f

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-091c10b5201

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b50bd8af7a7

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b50bd8af7a7

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-fc9d6770a55

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-f29c53cc267

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-176f0799bd9

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-176f0799bd9

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-176f0799bd9

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-cb318af614b

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-fb77357100d

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-4d71cbd5951

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-4d71cbd5951

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-4d71cbd5951

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-04dd8d32afc

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-0deac9a8d9a

benwtrent committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b062ed82417

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Add support for hiragana_uppercase & katakana_uppercase token filters in kuromoji analysis plugin (#106553)

This adds support for `hiragana_uppercase` and `katakana_uppercase` provided in the new lucene release.

Sutegana (捨て仮名) is small letter of hiragana and katakana in Japanese. In the old Japanese text, sutegana (捨て仮名) is not used unlikely to modern one. For example:

- "ストップウォッチ" is written as "ストツプウオツチ"
- "ちょっとまって" is written as "ちよつとまつて"

So it's meaningful to normalize sutegana to normal (uppercase) characters if we search against the corpus which includes old Japanese text such as patents, legal documents, contract policies, etc.

Related to: https://github.com/apache/lucene/pull/12915

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-6d908b814ba

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Test mute for issue #106683 (#106687)

relates: https://github.com/elastic/elasticsearch/issues/106683

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-6d908b814ba

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-da0c8b588b6

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-ae5d3534e3e

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-e177b28fccc

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-e177b28fccc

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-72439d29cc0

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-50a4475ee8d

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-34f1cce2722

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-34f1cce2722

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b9844481e3a

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot

jpountz committed:
Merge branch 'main' into lucene_snapshot

jpountz committed:
Merge branch 'main' into lucene_snapshot

jpountz committed:
Put MADV_RANDOM behind a feature flag. (#107280)

This puts `MADV_RANDOM` behind a feature flag to decouple this new feature from
Lucene from the Lucene upgrade itself. As a feature flag, snapshot builds will
have support for `MADV_RANDOM` (unless explicitly disabled) and release builds
will not support `MADV_RANDOM` (unless explicitly enabled).

breskeby committed:
Wrap aws operation in a retry for uploading due stability with dynamic creds

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-5ac65df4806

jpountz committed:
Fix compilation on lucene_snapshot branch after recent Lucene upgrade. (#107361)

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-5ac65df4806

ChrisHegarty committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot

ChrisHegarty committed:
Post merge fixes

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b727f0de92a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b727f0de92a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b727f0de92a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-fb978401b04

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Fixing failing tests for lucene_snapshot (#107437)

* Adding confidence_interval to one of the tests

* Fixing mapper testKnnQuantizedHNSWVectorsFormat

* Adding deterministic confidence interval for int8 flat

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b593924efe1

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-f52e7c708a3

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-f52e7c708a3

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-cb1d755b8a0

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-cb1d755b8a0

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-cb1d755b8a0

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-d8a81476262

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Fixing compilation due to lucene update

benwtrent committed:
spotless

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Fixing custom scalar codec scoring & DenseVectorFieldMapperTests (#107843)

* Fixing custom scalar codec scoring

* Fixing DenseVectorFieldMapperTests

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

ChrisHegarty committed:
Enable native access for Lucene madvise

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-9287167e351

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b70189819aa

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Fix compilation to account for new Lucene snapshot build (#108194)

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-166b47d45b3

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-26ec4c600d7

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-26ec4c600d7

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-26ec4c600d7

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-d8390866bd9

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b9457b70d37

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b9457b70d37

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b9457b70d37

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

mayya-sharipova committed:
Unified Highlighter to support matched_fields (#107640)

Add support to the Unified highlighter to combine matches on multiple fields
to highlight a single field: "matched_fields".

Based on Lucene PR: https://github.com/apache/lucene/pull/13268

Lucene PR is based on the concept of masked fields where masked fields
are different from the original highlighted field. This PR in
Elasticsearch uses the already existing highlighter parameter
"matched_fields".

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-b888652ff2e

benwtrent committed:
Fixing lucene snapshot compilation due to new VectorValue interface (#108516)

* Fixing lucene snapshot compilation due to new VectorValue interface

* fixing test compilation

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-3a4e4e33b0f

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-3a4e4e33b0f

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-3a4e4e33b0f

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-8580d50dbb9

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Fixing compilation with new Lucene API changes

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-a6d76fcb321

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-a30d7285cfb

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-91f45b53e6b

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-91f45b53e6b

ChrisHegarty committed:
Refactor custom ES814HnswScalarQuantizedVectorsFormat to use the FlatVectorsScorer API (#108576)

This commit removes the copy of the ScalarQuantizedVectorsWriter that we added in 8.14, in favour of a custom FlatVectorsScorer.

Previously we maintained the copy of the writer so that we could add a custom scorer implementation when merging scalar quantized vectors. This is not necessary now as we can provide a FlatVectorsScorer implementation that provides the custom scorer.

ChrisHegarty committed:
Re-enable comparisons in VectorScorerFactoryTests

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-5e48fdd0d45

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-5e48fdd0d45

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-5e48fdd0d45

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-26fca9e30c5

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-e913d5a7367

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-e913d5a7367

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-1c9fc15474a

ChrisHegarty committed:
Post merge cleanup

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-7f722d0ed67

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-7f722d0ed67

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-7f722d0ed67

ChrisHegarty committed:
Merge upstream/lucene_snapshot into lucene_snapshot

ChrisHegarty committed:
Use Default Flat Vectors Scorer singleton instance

ChrisHegarty committed:
Revert RandomAccessVectorValues to RandomAccessQuantizedByteVectorValues

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-0ba07db35ee

ChrisHegarty committed:
Fix lucene_snapshot build

ChrisHegarty committed:
Mute test until Lucene bug is fixed

ChrisHegarty committed:
Replace use of reflection with MemorySegmentAccessInput (#109061)

This commit replaces the use of reflection with the newly added MemorySegmentAccessInput.

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-67d80db41dd

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

ChrisHegarty committed:
Remove awaits fix

Commit:
[Automated] Update Lucene snapshot to 9.11.0-snapshot-1f02f961480

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

ChrisHegarty committed:
Use the SIMD optimized SQ vector scorer at search time (#109109)

This commit extends the custom SIMD optimized SQ vector scorer to include search time scoring.

When run on JDK22+ vector scoring with be done with the custom scorer. The implementation uses the JDK 22+ on-heap ALLOW_HEAP_ACCESS Linker.Option so that the native code can access the query vector directly.

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-c5ea94fbc6a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

benwtrent committed:
Update lucene snapshot to latest 9.11

benwtrent committed:
Fixing versions for Lucene 9.11

elasticsearchmachine commented:
Hi @benwtrent, I've created a changelog YAML for you.

benwtrent committed:
Update docs/changelog/109219.yaml

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

benwtrent committed:
updating to latest snapshot (RC1)

benwtrent committed:
Merge branch 'lucene_snapshot_9_11' of github.com:benwtrent/elasticsearch into lucene_snapshot_9_11

benwtrent commented:
@ChrisHegarty https://gradle-enterprise.elastic.co/s/xz2o7b5dz5dcu

This is the latest windows build failure. Seems like Windows & randomScorer stuff doesn't play nice?

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

ChrisHegarty commented:
> This is the latest windows build failure. Seems like Windows & randomScorer stuff doesn't play nice?

https://github.com/elastic/elasticsearch/pull/109303

ChrisHegarty committed:
Add a couple of missing supported checks to org.elasticsearch.vec.VectorScorerFactoryTests (#109303)

This commit adds a couple of missing supported checks to org.elasticsearch.vec.VectorScorerFactoryTests. Without these the test try, and fail, to run on Windows.

ChrisHegarty commented:
@elasticmachine run elasticsearch-ci/part-5-arm

benwtrent commented:
buildkite test this please

john-wagster committed:
Multivalue Sparse Vector Support (#109007)

Updated LuceneDocument to take advantage of looking up feature values on existing features and selecting the max when parsing multi-value sparse vectors

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

benwtrent committed:
Merge branch 'main' into lucene_snapshot_9_11

benwtrent committed:
Merge branch 'main' into lucene_snapshot_9_11

elasticsearchmachine commented:
Pinging @elastic/es-search (Team:Search)

ChrisHegarty commented on build-tools-internal/version.properties:
```diff
@@ -1,5 +1,5 @@
 elasticsearch     = 8.15.0
-lucene            = 9.10.0
+lucene            = 9.11.0-snapshot-d433394b292
```
might wanna flip this on the Lucene RC build (rather than a snapshot)

benwtrent replied:
OK, I tried this, and added the RC location as mvn location, but now ES is looking in that location for ALL packages, and its failing the build. Any ideas on how to simply add a secondary mvn location to look for packages instead of overriding?

benwtrent committed:
Use RC1 for release

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

mark-vieira commented on build.gradle:
```diff
@@ -239,6 +239,11 @@ if (project.gradle.startParameter.taskNames.any { it.startsWith("checkPart") ||
+
+  repositories {
+    // TODO: Temporary for Lucene RC builds. REMOVE
+    maven { url "https://dist.apache.org/repos/dist/dev/lucene/lucene-9.11.0-RC1-rev-d433394b292e3562e0bb34222f7dd4f307e2b8ca/lucene/maven" }
```
Is this still required now that we're pulling in the final version?

benwtrent replied:
@mark-vieira this isn't the final version, this is the RC build for the final version. 

I wanted to test all our BWC, etc. against the final build, but alas, now everything is pulling from this url and tests are broken. 

Is there a way for me to update this as an additional optional place to look for binaries so that Elasticsearch can see the release candidate for Lucene?

mark-vieira replied:
It looks like those failures are because dist.apache.org is unreachable. Some kind of transient network issue? In any case, we can restrict this to say "only use this repo for lucene dependencies":

```groovy
repositories {
  exclusiveContent {
    forRepository {
      maven { url "https://dist.apache.org/repos/dist/dev/lucene/lucene-9.11.0-RC1-rev-d433394b292e3562e0bb34222f7dd4f307e2b8ca/lucene/maven" }
    }
  }
  filter {
    includeGroup "org.apache.lucene"
  }
}
```

benwtrent committed:
adjusting RC1 build url and adding release note highlight

benwtrent committed:
Removing release-candidate staging repo

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

iverase reviewed (commented):
I am investigating a possible issue in lucene 9.11, please do not merge.

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11

benwtrent committed:
Bypassing intra-merge threading for now

benwtrent commented:
While still investigating here: https://github.com/apache/lucene/pull/13475

I have disabled intra-merge concurrency by always returning a same thread executor.

benwtrent commented:
@martijnvg all the failures are around the new dynamic setting being changed. 

```
Likely root cause: java.lang.IllegalArgumentException: Setting index.mapper.dynamic was removed after version 6.0.0 |  
-- | --
  | at org.elasticsearch.index.mapper.MapperService.<init>(MapperService.java:208) |  
  | at org.elasticsearch.cluster.metadata.IndexMetadataVerifier.checkMappingsCompatibility(IndexMetadataVerifier.java:177) |  
  | at org.elasticsearch.cluster.metadata.IndexMetadataVerifier.verifyIndexMetadata(IndexMetadataVerifier.java:87) |  
  | at org.elasticsearch.gateway.GatewayMetaState.upgradeMetadata(GatewayMetaState.java:279) |  
  | at org.elasticsearch.gateway.GatewayMetaState.upgradeMetadataForNode(GatewayMetaState.java:265) |  
  | at org.elasticsearch.gateway.GatewayMetaState.start(GatewayMetaState.java:178) |  
  | at org.elasticsearch.node.Node.start(Node.java:1187) |  
  | at org.elasticsearch.bootstrap.Bootstrap.start(Bootstrap.java:335) |  
  | at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:443) |  
  | at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:166) |  
  | at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:157) |  
  | at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:77) |  
  | at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:112) |  
  | at org.elasticsearch.cli.Command.main(Command.java:77) |  
  | at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:122) |  
  | at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:80)
```

Now, I am not sure why these particular BWC tests are running against much older versions than 7.17.22, but they are.

Here is an example failure: https://gradle-enterprise.elastic.co/s/7wyrf5fogxcqq

martijnvg commented:
@benwtrent Looks like the test ran against `7.12.1`, this upgrade test should only run with 7.17.latest as old version. I will update this test to skip if old version is any other version.

martijnvg commented:
I opened #109574

benwtrent commented:
@martijnvg I realize that it ran against old versions and its weird to me, but there must be a reason? I am pretty sure versions have to be specifically set somewhere.

Do we know why these old versions are being tested against?

martijnvg commented:
> Do we know why these old versions are being tested against?

I don't know. I actually surprised like you too. The PR that added these tests didn't have a `test-full-bwc` label, because I thought we were only testing against 7.17.latest for 7.x bwc tests and not against older 7.x minors. When I saw this test failure, I thought my assumption was just wrong. But right now I'm not sure anymore.

martijnvg commented:
We do document that from 7.x to 8.x, a cluster must first be on 7.17.x before doing the actual 8.x upgrade: https://www.elastic.co/guide/en/elastic-stack/8.14/upgrading-elastic-stack.html#prepare-to-upgrade

So I don't think we need to test upgrading from 7.16 or earlier to 8.x?

benwtrent commented:
@martijnvg I am not 100% a PR gets tested against 7.17 at all unless the `test-full-bwc` tag is added :/

mark-vieira commented:
> We do document that from 7.x to 8.x, a cluster must first be on 7.17.x before doing the actual 8.x upgrade: https://www.elastic.co/guide/en/elastic-stack/8.14/upgrading-elastic-stack.html#prepare-to-upgrade
> 
> So I don't think we need to test upgrading from 7.16 or earlier to 8.x?

We do, because we test the test path you describe. So we don't do rolling upgrade tests, but we still run full cluster restart tests that start a 7.16 cluster, upgrade to 7.17, and then upgrade to the current version.

> @martijnvg I am not 100% a PR gets tested against 7.17 at all unless the test-full-bwc tag is added :/

Every PR is tested against the HEAD of the 7.17 branch, even without that label. The label will add testing for previously released version of Elasticsearch.

benwtrent committed:
Merge remote-tracking branch 'upstream/main' into lucene_snapshot_9_11
